pkg/kubelet/kubeletconfig/configsync_test.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeletconfig
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	api "k8s.io/kubernetes/pkg/apis/core"
+)
+
+func TestLatestNodeConfigSource(t *testing.T) {
+
+	indexer := cache.NewStore(cache.MetaNamespaceKeyFunc)
+
+	configResource := &apiv1.NodeConfigSource{
+		ConfigMap: &apiv1.ConfigMapNodeConfigSource{
+			Name:             "c-name",
+			Namespace:        "c-namespace",
+			UID:              "c-uid",
+			KubeletConfigKey: "c-key",
+		},
+	}
+
+	node := &apiv1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "node",
+			UID:  "node-uid",
+		},
+		Spec: apiv1.NodeSpec{
+			ConfigSource: configResource,
+		},
+	}
+	_ = indexer.Add(node)
+
+	nodeApi := &api.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "nodeApi",
+			UID:  "nodeApi-uid",
+		},
+	}
+	_ = indexer.Add(nodeApi)
+
+	tests := []struct {
+		NodeName  string
+		expectErr error
+		expect    *apiv1.NodeConfigSource
+	}{
+		{
+			NodeName:  "node",
+			expectErr: nil,
+			expect:    configResource,
+		},
+		{
+			NodeName:  "node-none",
+			expectErr: fmt.Errorf("node %q does not exist in the informer's store, can't sync config source", "node-none"),
+			expect:    nil,
+		},
+		{
+			NodeName:  "nodeApi",
+			expectErr: fmt.Errorf("failed to cast object from informer's store to Node, can't sync config source for Node %q", "nodeApi"),
+			expect:    nil,
+		},
+	}
+
+	for _, test := range tests {
+		actual, err := latestNodeConfigSource(indexer, test.NodeName)
+		if err == nil {
+			if test.expect == actual || !reflect.DeepEqual(test.expect, actual) {
```

Comment:
This looks confusing!! Why checking using both `==` and reflect?
 please print expected and actual value in log.

Reply:
ok